packages/store/ts/config/v2/scope.ts

```diff
@@ -0,0 +1,36 @@
+import { Dict, evaluate } from "@arktype/util";
+import { SchemaInput } from "./schema";
+import { StaticAbiType, DynamicAbiType } from "@latticexyz/schema-type";
+
+export type AbiType = StaticAbiType | DynamicAbiType;
+
+export const EmptyScope = { validTypes: {} } as const satisfies ScopeOptions;
+export type EmptyScope = typeof EmptyScope;
+
+export type AbiTypeScope = ScopeOptions<{ [t in AbiType]: t }>;
+export const AbiTypeScope = { validTypes: {} } as AbiTypeScope; // TODO: runtime implementation
+
+export type ScopeOptions<validTypes extends Dict<string, AbiType> = Dict<string, AbiType>> = {
+  validTypes: validTypes;
```

Comment:
would this ever include invalid types? 🤔

Reply:
renamed to `types`